src/libs/actions/IOU.ts

```diff
@@ -5554,7 +5554,24 @@ function deleteTrackExpense(chatReportID: string, transactionID: string, reportA
         return deleteMoneyRequest(transactionID, reportAction, isSingleTransactionView);
     }
 
-    const {parameters, optimisticData, successData, failureData, shouldDeleteTransactionThread} = getDeleteTrackExpenseInformation(chatReportID, transactionID, reportAction);
+    const chatReportActions = allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReportID}`] ?? {};
+    const whisperAction = Object.values(chatReportActions).find(
+        (action: OnyxTypes.ReportAction) =>
+            action?.originalMessage &&
+            action?.actionName === CONST.REPORT.ACTIONS.TYPE.ACTIONABLE_TRACK_EXPENSE_WHISPER &&
+            action.originalMessage.transactionID === transactionID &&
+            ReportActionsUtils.isActionableTrackExpense(action),
+    );
+    const actionableWhisperReportActionID = whisperAction?.reportActionID;
+    const {parameters, optimisticData, successData, failureData, shouldDeleteTransactionThread} = getDeleteTrackExpenseInformation(
+        chatReportID,
+        transactionID,
+        reportAction,
+        undefined,
+        undefined,
+        actionableWhisperReportActionID,
```

Comment:
The following test steps will result in unexpected behavior if API request fails. 

1. Go Offline.
2. Navigate to Self DM.
3. Create a new Track Expense (Verify that the Track Expense is created along with the Actionable Whisper)
4. Select `Nothing for now` in Actionable Whisper
5. Open the menu for Track Expense Report Action
6. Click/Tap on `Delete expense`
7. Go Online

When `Delete expense` API request fails, 
Actual Behaviour: Since we reset `resolution` [here](https://github.com/Expensify/App/blob/af9be0aaf6a00eed6e884ae84bc7f8509ec2bac0/src/libs/actions/IOU.ts#L1695), the Actionable Whisper will get shown which is not correct. 
Expected Behaviour: The Track Expense Actionable Whisper should be hidden as we had already selected `Nothing for now` in step 4. 

So, instead of passing the `actionableWhisperReportActionID` here, we can fetch the Track Expense Actionable Whisper within `getDeleteTrackExpenseInformation` itself and determine the `actionableWhisperReportActionID`. 
Further, we can know the existing `resolution` value within Track Expense Actionable Whisper which can be used to reset `resolution` on failure [here](https://github.com/Expensify/App/blob/af9be0aaf6a00eed6e884ae84bc7f8509ec2bac0/src/libs/actions/IOU.ts#L1695). 
Would this work?